keras_preprocessing/image/dataframe_iterator.py

```diff
@@ -49,16 +49,19 @@ class DataFrameIterator(Iterator):
             Color mode to read images.
         classes: Optional list of strings, classes to use (e.g. `["dogs", "cats"]`).
             If None, all classes in `y_col` will be used.
-        class_mode: Mode for yielding the targets:
-            `"binary"`: binary targets (if there are only two classes),
-            `"categorical"`: categorical targets, supports multi-label output
-                if class values per observation are stored in a list or tuple
-                in the `y_col` column,
-            `"sparse"`: integer targets,
-            `"input"`: targets are images identical to input images (mainly
-                used to work with autoencoders),
-            `"other"`: targets are the data(numpy array) of y_col data
-            `None`: no targets get yielded (only input images are yielded).
+        class_mode: one of "categorical", "binary", "sparse", "input",
+            "other" or None. Default: "categorical".
+            Mode for yielding the targets:
+            - `"binary"`: 1D numpy array of binary labels,
+            - `"categorical"`: 2D numpy array of one-hot encoded labels.
+                Supports multi-label output.
+            - `"sparse"`: 1D numpy array of integer labels,
+            - `"input"`: images identical to input images (mainly used to
+                work with autoencoders),
+            - `"other"`: numpy array of `y_col` data,
+            - `None`, no targets are returned (the generator will only yield
+                batches of image data, which is useful to use in
+            `model.predict_generator()`, `model.evaluate_generator()`, etc.).
```

Comment:
`model.evaluate_generator()` takes both x and y data and returns loss by comparing it with predictions, So I don't think it will work with `class_mode=None`

Reply:
are you sure? because I also see it in `fir_from_directory` docstring

https://github.com/keras-team/keras-preprocessing/blame/master/keras_preprocessing/image/image_data_generator.py#L484

Reply:
Evaluate generator does need y data, refer the while loop statement here(https://github.com/keras-team/keras/blob/1336cdb14ff03de754aec6899794742ca91057b2/keras/engine/training_generator.py#L346)
If class_mode=None, y data is not returned right?

Reply:
You are correct. Thanks for pointed it out. I'll make the changes in both places.

Reply:
is it OK now? please let me know if you notice something else.